apps/emqx_auth/src/emqx_authz/emqx_authz.erl

```diff
@@ -499,33 +499,57 @@ do_authorize(
     Module = authz_module(Type),
     emqx_metrics_worker:inc(authz_metrics, Type, total),
     try Module:authorize(Client, PubSub, Topic, Connector) of
+        %% for example emqx_authz_client_info without acl rule.
+        %% skip client tracing when client_info returns "skip" in non-superuser mode.
+        %% this trace log cause user confusion.
+        skip ->
+            do_authorize(Client, PubSub, Topic, Tail);
         nomatch ->
             emqx_metrics_worker:inc(authz_metrics, Type, nomatch),
-            ?TRACE("AUTHZ", "authorization_module_nomatch", #{
+            ?TRACE("AUTHZ", "authorization_nomatch", #{
+                authorize_type => Type,
+                module => Module,
+                username => Username,
+                topic => Topic,
+                action => emqx_access_control:format_action(PubSub)
+            }),
+            do_authorize(Client, PubSub, Topic, Tail);
+        ignore ->
+            ?TRACE("AUTHZ", "authorization_ignore", #{
+                authorize_type => Type,
                 module => Module,
                 username => Username,
                 topic => Topic,
                 action => emqx_access_control:format_action(PubSub)
             }),
             do_authorize(Client, PubSub, Topic, Tail);
-        %% {matched, allow | deny | ignore}
         {matched, ignore} ->
-            ?TRACE("AUTHZ", "authorization_module_match_ignore", #{
+            ?TRACE("AUTHZ", "authorization_matched_ignore", #{
+                authorize_type => Type,
                 module => Module,
                 username => Username,
                 topic => Topic,
                 action => emqx_access_control:format_action(PubSub)
             }),
             do_authorize(Client, PubSub, Topic, Tail);
-        ignore ->
-            ?TRACE("AUTHZ", "authorization_module_ignore", #{
+        {matched, allow} ->
+            ?TRACE("AUTHZ", "authorization_matched_allow", #{
+                authorize_type => Type,
+                module => Module,
+                username => Username,
+                topic => Topic,
+                action => emqx_access_control:format_action(PubSub)
+            }),
+            do_authorize(Client, PubSub, Topic, Tail);
+        {matched, deny} ->
+            ?TRACE("AUTHZ", "authorization_matched_deny", #{
+                authorize_type => Type,
```

Comment:
we also trace the final result. Otherwise, we wouldn't know which authz was in effect.